[Update] Update the ili210x support for ilitek ili251x touch device

samhsuism opened the pull request:
The upstream's ili210x touchscreen driver is updated since v5.1-rc1, this latest version of ili210x driver has been compiled and tested with the new changes of the ili210.c source and the relative necessary edits such as the device tree, the config files ( in bcmrpi_defconfig, bcm2709_defconfig ) and the Makefile. It works as functional and we would like these new patches could be updated to the next released Raspberry Pi's image for the supports of the Ilitek touch devices ili210x and ili251x.

samhsuism committed:
Update ili210x.c

samhsuism committed:
overlays: Add ilitek251x

samhsuism committed:
BCM2708: Add core Device Tree support, ilitek251x

samhsuism committed:
overlays: Add ilitek251x overlay

samhsuism committed:
configs: Add TOUCHSCREEN_ILI210X=m

samhsuism committed:
configs: Add TOUCHSCREEN_ILI210X=m

samhsuism committed:
Input: ilitek251x - add support for ilitek251x

samhsuism committed:
Update Makefile

pelwell commented:
If some of these commits are (or contain) upstream changes as you suggest, please include those commits without modification (except for adding `commit ... upstream.` as the first line of the comment, where `...` is replaced with the full commit hash), followed by a blank line.

samhsuism committed:
Input: ili210x - fetch touchscreen geometry from DT

Fetching the geometry from the ILI251x registers seems unreliable and
sometimes returns all zeroes. Add support for fetching the geometry and
axis inversion from DT instead.

Signed-off-by: Marek Vasut <marex@denx.de>
Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>
